src/compiler/checker.ts

```diff
@@ -29504,6 +29504,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return !isPastLastAssignment(symbol, /*location*/ undefined);
     }
 
+    // Check if a parameter or catch variable is assigned definitely

```

Comment:
This comment (and the one for `isSymbolAssigned`) is not accurate, I think, because we also check if mutable local variables are assigned definitely.